rclcpp/include/rclcpp/callback_group.hpp

```diff
@@ -97,6 +98,8 @@ class CallbackGroup
   add_client(const rclcpp::client::ClientBase::SharedPtr client_ptr);
 
   CallbackGroupType type_;
+  // Mutex to protect the subsequent vectors of pointers.
+  mutable std::mutex lock_;
```

Comment:
nitpick: this is a mutex not a lock, maybe it should be called `mutex_`? I don't consider this a blocker to merge.